fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -1509,6 +1509,144 @@ void simpleCrossPartitionQuery() {
+                    Type.LESS_THAN_OR_EQUALS, Map.of(SortType.ASCENDING, 0, SortType.DESCENDING, 2, SortType.UNSORTED, 2),
+                    Type.EQUALS, Map.of(SortType.ASCENDING, 2, SortType.DESCENDING, 2, SortType.UNSORTED, 2));
+            startingPartitionExpectation.put(time1_2, expectationsForTime1_2);
+
+            // =====================================
+            // Expectations for timeTooOld
+            // =====================================
+            Map<Comparisons.Type, Map<SortType, Integer>> expectationsForTimeTooOld = Map.of(
+                    Type.GREATER_THAN, Map.of(SortType.ASCENDING, 0, SortType.DESCENDING, 3, SortType.UNSORTED, 3),
+                    Type.GREATER_THAN_OR_EQUALS, Map.of(SortType.ASCENDING, 0, SortType.DESCENDING, 3, SortType.UNSORTED, 3),
+                    Type.LESS_THAN, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1),
+                    Type.LESS_THAN_OR_EQUALS, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1),
+                    Type.EQUALS, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1));
+            startingPartitionExpectation.put(timeTooOld, expectationsForTimeTooOld);
+
+            // =====================================
+            // Expectations for timeTooNew
+            // =====================================
+            Map<Comparisons.Type, Map<SortType, Integer>> expectationsForTimeTooNew = Map.of(
+                    Type.GREATER_THAN, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1),
+                    Type.GREATER_THAN_OR_EQUALS, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1),
+                    Type.LESS_THAN, Map.of(SortType.ASCENDING, 0, SortType.DESCENDING, 3, SortType.UNSORTED, 3),
+                    Type.LESS_THAN_OR_EQUALS, Map.of(SortType.ASCENDING, 0, SortType.DESCENDING, 3, SortType.UNSORTED, 3),
+                    Type.EQUALS, Map.of(SortType.ASCENDING, -1, SortType.DESCENDING, -1, SortType.UNSORTED, -1));
+            startingPartitionExpectation.put(timeTooNew, expectationsForTimeTooNew);
+
+            // check all combinations against expectations
+            for (Long predicateComparand : List.of(time0, time1, time2, time1_2, timeTooOld, timeTooNew)) {
+                for (Comparisons.Type comparisonType : List.of(Type.GREATER_THAN, Type.GREATER_THAN_OR_EQUALS, Type.LESS_THAN, Type.LESS_THAN_OR_EQUALS, Type.EQUALS)) {
+                    for (SortType sortType : EnumSet.allOf(SortType.class)) {
+                        LOGGER.debug("comparison: {} sort: {} time: {}", comparisonType, sortType, timesForLogging.get(predicateComparand));
+                        LuceneScanQuery luceneScanQuery = buildLuceneScanQuery(index, isSynthetic, comparisonType, sortType, predicateComparand, luceneSearch);
```

Comment:
It would be good to also have some tests of the logic that checks whether a predicate is suitable.
Perhaps it would be worth making that logic package-level, and then having tests that plan a query and assert that it either has a predicate or doesn't have a predicate, rather than asserting about which partition is chosen.